dcrsqlite/apisource.go

```diff
@@ -289,20 +289,41 @@ func (db *wiredDB) GetAllTxOut(txid string) []*apitypes.TxOut {
 // GetRawTransactionWithPrevOutAddresses looks up the previous outpoints for a
 // transaction and extracts a slice of addresses encoded by the pkScript for
 // each previous outpoint consumed by the transaction.
-func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes.Tx, [][]string) {
+func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes.Tx, [][]string, string) {
 	tx := db.GetRawTransaction(txid)
 	prevOutAddresses := make([][]string, len(tx.Vin))
 	if tx == nil {
-		return tx, prevOutAddresses
+		return nil, prevOutAddresses, ""
+	}
+	txhash, err := chainhash.NewHashFromStr(txid)
+	if err != nil {
+		log.Errorf("Invalid transaction hash %s", txid)
+		return nil, prevOutAddresses, ""
 	}
 
+	txR, err := db.client.GetRawTransaction(txhash)
```

Comment:
Would be best to avoid a second call to dcrd.  We can refactor a little to pass the transaction bytes back:

```go
func (db *wiredDB) GetRawTransaction(txid string) *apitypes.Tx {
	tx, _ := db.getRawTransaction(txid)
	return tx
}

func (db *wiredDB) getRawTransaction(txid string) (*apitypes.Tx, string) {
	tx := new(apitypes.Tx)
...
```
then
```go
func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes.Tx, [][]string, string) {
	tx, txhex := db.getRawTransaction(txid)
...
```

And then we can get the `MsgTx` by decoding the hex string representation of the tx:

```go
	txBytes, err := hex.DecodeString(txhex)
	if err != nil {
		log.Errorf("Unable to read tx %v: %v", txid, err)
		return nil, nil, ""
	}
	msgTx := wire.NewMsgTx()
	if err = msgTx.FromBytes(txBytes); err != nil {
		log.Errorf("Failed to deserialize MsgTx: %v", err)
		return nil, nil, ""
	}
```

This last snippet could be a function in txhelpers.